### PR TITLE
[14.0][FIX+IMP] storage_image, storage_image_product and storage_image_product_brand

### LINKED ISSUE #95: [14.0] Storage Image function is broken
In V14 an error occurs while trying to ad an image record.

Steps to reproduce
Go to Settings->Technical->Storage backend -> Image

The widget for uploading an image is unavalable.
When saving  an record an error occurs
<details>

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1389, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1381, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 394, in call_kw
    result = _call_kw_model_create(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 374, in _call_kw_model_create
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-162>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 323, in _model_create_single
    return create(self, arg)
  File "/home/odoo/build/OCA/storage/setup/storage_image/odoo/addons/storage_image/models/storage_image.py", line 53, in create
    image = super(StorageImage, self).create(vals)
  File "<decorator-gen-159>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 323, in _model_create_single
    return create(self, arg)
  File "/home/odoo/build/OCA/storage/setup/storage_thumbnail/odoo/addons/storage_thumbnail/models/thumbnail_mixin.py", line 71, in create
    record = super().create(vals)
  File "<decorator-gen-65>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 344, in _model_create_multi
    return create(self, [arg])
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_fields.py", line 533, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 345, in _model_create_multi
    return create(self, arg)
  File "/.repo_requirements/odoo/odoo/models.py", line 3862, in create
    for data in parent_data_list
  File "<decorator-gen-65>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 345, in _model_create_multi
    return create(self, arg)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_fields.py", line 533, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/.repo_requirements/odoo/odoo/api.py", line 345, in _model_create_multi
    return create(self, arg)
  File "/.repo_requirements/odoo/odoo/models.py", line 3901, in create
    fields[0].determine_inverse(batch_recs)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1185, in determine_inverse
    getattr(records, self.inverse)()
  File "/home/odoo/build/OCA/storage/setup/storage_file/odoo/addons/storage_file/models/storage_file.py", line 118, in _inverse_data
    record.write(record._prepare_meta_for_file())
  File "/home/odoo/build/OCA/storage/setup/storage_file/odoo/addons/storage_file/models/storage_file.py", line 107, in _prepare_meta_for_file
    bin_data = base64.b64decode(self.data)
  File "/usr/lib/python3.6/base64.py", line 80, in b64decode
    s = _bytes_from_decode_data(s)
  File "/usr/lib/python3.6/base64.py", line 46, in _bytes_from_decode_data
    "string, not %r" % s.__class__.__name__) from None
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: argument should be a bytes-like object or ASCII string, not 'bool'
```

</details>

image

Expected behaviour: Like in V12.
Widget for image uploading is displayed.
image

### PR DESCRIPTION
**Pull request by ivantodorovich**
Closes https://github.com/OCA/storage/issues/95

* Improve image_handle widgets drag&drop feature.
* Refactor code + remove dead code.
* New and improved base views for image.relation.abstract
* Improve tests

### CODE REVIEW

**Comment by ivantodorovich** [comment]
Actually no, still working on it :man_facepalming:

**Comment by ivantodorovich** [comment]
Regarding: https://github.com/OCA/storage/issues/95

This thing played tricks on me.. at first I couldn't understand why it worked perfectly fine with `debug=assets`, but not without it.. then I realized odoo added their own `image_url` widget and it was conflicting with ours, depending on the script loading order :man_facepalming: 

So I ended up removing the widget alltogether. It doesn't allow to edit pictures when using it, but that's ok.. storage images are not meant to be edited. For create, we can simply use the image widget on the `data` field. So, on the `storage.image` form view, we're not using the `image_url` widget anymore.. anyway, the user doesn't go there often so it's a good tradeoff I think. Also this way we can remove the ugly hack on `storage.image::create`

I also renamed `image_handle` to `storage_image_handle` to avoid possible issues like this one

**Comment by ivantodorovich** [comment]
Also fixed `storage_image_import` tests that were failing after my refactor and didn't make much sense to me.
Rewrote them in a simpler way IMO, here: https://github.com/OCA/storage/pull/103/commits/717617b646525ee53aebb98d479558cf2a9a4604 also removing the dependency to `storage_image_product` that was only there for the previous tests cases

**Comment by simahawk** [comment]
Please fwd port this before doing these changes https://github.com/OCA/storage/pull/94

File: storage_image_import/tests/models/fake_image_relation.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2021 Camptocamp (http://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class FakeImageRelation(models.Model):
```
Comment (simahawk) [inline storage_image_import/tests/models/fake_image_relation.py:8]: see how I used odoo-test-helper https://github.com/OCA/storage/pull/94 (to be fwd ported)

**Review by simahawk — CHANGES_REQUESTED** [review]
pls split the image widget fix

File: storage_image/security/security.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
```
Comment (simahawk) [inline storage_image/security/security.xml]: what's the purpose of merging 2 records in a single file? It's preferable to have the in 2 files as before.

**Review by simahawk — COMMENTED** [review]
LG, can you restore the security files?

File: storage_image/security/security.xml (hunk 2 above)
Comment (ivantodorovich) [reply storage_image/security/security.xml]: Oh, didn't think much about it. It's just standardization, as this is how you'd find it in the majority of modules.
I'll revert if you mind it.

Comment (ivantodorovich) [reply storage_image/security/security.xml]: done!

**Review by simahawk — APPROVED** [review]

**Comment by sebastienbeau** [comment]
Hi @ivantodorovich

Thanks for the refactor, opening the pop-up is really better.

The only missing feature with the previous version is the feedback during the drag and drop
image

Indeed when you drop the first image the zone where the drag and drop work is not so big and without feedback you can drop in in the wrong place and this will open the image in the browser

**Comment by sebastienbeau** [comment]
@ivantodorovich the weird poijnt is that you have defined a css for the drop-zone here https://github.com/OCA/storage/pull/103/files#diff-0388f7ef60e004bc536459637a780f8aa1261f05ce7f0d8f0a8f383308e46cf0R9

**Comment by ivantodorovich** [comment]
> @ivantodorovich the weird poijnt is that you have defined a css for the drop-zone here https://github.com/OCA/storage/pull/103/files#diff-0388f7ef60e004bc536459637a780f8aa1261f05ce7f0d8f0a8f383308e46cf0R9

Indeed, it seems I broke it by mistake.
I've just pushed the fix :) Thanks for catching it!

**Review by sebastienbeau — APPROVED** [review]
LGTM code review and test.
Thanks for the improvement and the quick fix

**Comment by sebastienbeau** [comment]
Now we just need to export the brand image in shopinvader module and we are done ;)

**Pull request merged**